File: problems/medium/_1679_max_number_of_k_sum_pairs/go/solution.go
package maxnumberofksumpairs

import "sort"

func maxOperations(nums []int, k int) int {
	sort.Ints(nums)

	left := 0
	right := len(nums) - 1
	operation := 0

	for left < right {
		sum := nums[left] + nums[right]
		if sum == k {
			left++
			right--
			operation++
		} else if sum > k {
			right--
		} else {
			left++
		}
	}

	return operation
}

File: problems/medium/_1679_max_number_of_k_sum_pairs/kotlin/Solution.kt
package medium._1679_max_number_of_k_sum_pairs.kotlin

class Solution {
    fun maxOperations(nums: IntArray, k: Int): Int {
        nums.sort()

        var left = 0
        var right = nums.lastIndex
        var operation = 0;

        while (left < right) {
            val sum = nums[left] + nums[right]

            when {
                sum == k -> {
                    left++
                    right--
                    operation++
                }

                sum > k -> {
                    right--
                }

                else -> {
                    left++
                }
            }
        }
        return operation
    }
}